test/src/e2e_vm_tests/test_programs/should_pass/require_contract_deployment/storage_access_caller/src/main.sw

```diff
@@ -0,0 +1,168 @@
+script;
+use storage_access_abi::{S, StorageAccess, T};
+use std::chain::*;
+
+fn main() -> bool {
+    let contract_id = 0x35248b197c9f5ba30a2dfc20414c508efcd1bc4ad110efb26ac187fe81f1e57b;
+    let caller = abi(StorageAccess, contract_id);
+
+    // Test 1
+    caller.set_x {
+        gas: 10000
```

Comment:
Is there a reason to specify gas here instead of default? My concern is that if we set the gas cost to a storage write to e.g. 20k like on Ethereum, then all these tests will fail. Why not just forward all gas?

Reply:
`$cgas` _starts_ at `tx.gasLimit`, yes. But with each contract call, it's initially the gas forwarded. Note that if you try to forward more than `$cgas`, then only `$cgas` is forwarded. It should not be a panic in the VM.